profiler/src/ProfilerEngine/Datadog.Profiler.Native/ManagedThreadList.cpp

```diff
@@ -261,6 +261,7 @@ HRESULT ManagedThreadList::TryGetCurrentThreadInfo(std::shared_ptr<ManagedThread
         return E_FAIL;
     }
 
+    std::lock_guard<std::recursive_mutex> lock(_mutex);
     pThreadInfo = FindByClrId(clrThreadId);
```

Comment:
Just out of curiosity, since `_mutex` is `std::recursive_mutex` why aren't you locking from within `FindByClrId`?
since it's a _recursive_ mutex, you won't face a deadlock by locking it twice by the same thread.
🤔

Reply:
Thanks for the review.
Good point and you are right. we would like to review the locking strategy of this class at some point and went simple on the fix.

Reply:
Sounds good. Resolving.